plugins/debuginfo-install.py

```diff
@@ -74,59 +76,170 @@ def configure(self):
+                            self.installed_debugsource_missing.add(str(package))
+
+                    continue
+
+                # if the package in question is -debuginfo or -debugsource, install it directly
+                if first_package.name.endswith(Package.DEBUGINFO_SUFFIX) \
+                        or first_package.name.endswith(Package.DEBUGSOURCE_SUFFIX):
+
+                    self._install((first_package, ))
+                    continue
+
+                # if we have NEVRA parsed from the pkgspec, use it to install the package
+                if solution["nevra"] is not None:
+                    if not self._install_debug(first_package.debug_name, solution["nevra"]):
+                        self.available_debuginfo_missing.add(
+                            "{}-{}".format(first_package.name, first_package.evr))
+
+                    if not self._install_debug(first_package.debugsource_name, solution["nevra"]):
+                        self.available_debugsource_missing.add(
+                            "{}-{}".format(first_package.name, first_package.evr))
+
+                    continue
+
+                # inf we don't have NEVRA from the pkgspec, pass nevras from
+                # all packages that were found (while replacing the name with
+                # the -debuginfo and -debugsource variant) to the solver, which
+                # will pick the correct version and architecture
+                if not self._install_debug_no_nevra(first_package.debug_name, packages):
+                    self.available_debuginfo_missing.add(
+                        "{}-{}".format(first_package.name, first_package.evr))
+
+                if not self._install_debug_no_nevra(first_package.debugsource_name, packages):
+                    self.available_debugsource_missing.add(
+                        "{}-{}".format(first_package.name, first_package.evr))
+
+        if self.available_debuginfo_missing:
+            logger.info(
+                _("Could not find debuginfo package for the following available packages: %s"),
+                ", ".join(sorted(self.available_debuginfo_missing)))
+
+        if self.available_debugsource_missing:
+            logger.info(
+                _("Could not find debugsource package for the following available packages: %s"),
+                ", ".join(sorted(self.available_debugsource_missing)))
+
+        if self.installed_debuginfo_missing:
+            logger.info(
+                _("Could not find debuginfo package for the following installed packages: %s"),
+                ", ".join(sorted(self.installed_debuginfo_missing)))
+
+        if self.installed_debugsource_missing:
+            logger.info(
+                _("Could not find debugsource package for the following installed packages: %s"),
+                ", ".join(sorted(self.installed_debugsource_missing)))
 
         if errors_spec and self.base.conf.strict:
```

Comment:
Why missing installed_debugsource_missing does not trigger an error? Then we can remove one loop above (`for name in list(package_dict.keys()):`).

Reply:
I don't understand what you mean...

Reply:
What about to replace it by:
`if (errors_spec or self.installed_debugsource_missing) and self.base.conf.strict:`

I suggest that this will resolve the issue when debuginfo is present but not in version of installed package and argument like (foo*) is used.

Reply:
But how will it resolve the issue? This is just reporting an error at the end, how can it resolve the issue?

Also you've picked one of the four lists of missing packages, so I just don't see how that makes sense?

(I've pushed fixes for the minor comments you had)

Reply:
@j-mracek,  you have probably missed that based on the discussion in [bug 1586059](https://bugzilla.redhat.com/show_bug.cgi?id=1586059), there should be error only when the pkgspec doesn't match anything. If -debuginfo/-debugsource package is missing, it's not an error, because not all packages must have debuginfo. See [comment 15](https://bugzilla.redhat.com/show_bug.cgi?id=1586059#c15) for a reasoning.

Reply:
(I somehow thought the discussion is still about the `for name in list(package_dict.keys()):` "hack" way earlier in the code, also discussed in the thread above. Thanks @pkratoch)